packages/server-wallet/e2e-test/receiver/controller.ts

```diff
@@ -25,18 +24,23 @@ export default class ReceiverController {
   }
 
   public async acceptMessageAndReturnReplies(message: Payload): Promise<Payload> {
-    const {signedStates, objectives} = message;
+    const reply: Payload = {
+      signedStates: [],
+      objectives: [],
+    };
 
     const {
       channelResults: [channelResult],
+      outbox: [maybeSyncStateResponse],
```

Comment:
I'm not so sure about the naming here: the receiver controller shouldn't care the contents of the message -- it should just send the message.

That's what this code looks to do, but the name might be a bit deceptive. It also obscures the fact that there may be _multiple_ outgoing messages, and they should all be send to the relevant peer.

Reply:
Right now the receiver client is built for the end to end test and the only case where there would be anything in the outbox after a `pushMessage` call in the tests is the case where the payer sends a request for channel data (i.e., syncing). If we decided to add more features to the receiver then we would have to remove some of the hard coded elements of it (like this example where it's hard coded to expect a sync response in index 0 of the outbox, or nothing at all). However, then we start getting into the territory of writing a receipt manager. I guess a question would be how generalized should this test harness be?